### PR TITLE
xdynamic bitset refactor

### PR DESCRIPTION
**Pull request by wolfv**
Some changes that should have been integrated a while ago :)

Necessary to create bitset views, e.g. for Arrow buffers.

### CODE REVIEW

File: include/xtl/xdynamic_bitset.hpp [hunk 1]
```diff
@@ -811,136 +963,118 @@ namespace xtl
...
-    inline bool xdynamic_bitset<B, A>::operator>(const self_type& rhs) const noexcept
-    {
-        return rhs < *this;
-    }
-
-    template <class B, class A>
-    inline bool xdynamic_bitset<B, A>::operator>=(const self_type& rhs) const noexcept
-    {
-        return rhs <= *this;
-    }
```
Comment (JohanMabille) [inline include/xtl/xdynamic_bitset.hpp]: That's a backward incompatible change, so we would have to release xtl 0.5.0

File: include/xtl/xdynamic_bitset.hpp [hunk 2]
```diff
@@ -152,37 +186,95 @@ namespace xtl
...
-        storage_type m_buffer;
-        size_type m_size;
+        // Make views and buffers friends
+        template<typename BB>
+        friend class xdynamic_bitset_base;
     };
 
-    template <class B, class A>
-    xdynamic_bitset<B, A> operator~(const xdynamic_bitset<B, A>& lhs);
+    // NOTE this view ZEROS out remaining bits!
```
Comment (JohanMabille) [inline include/xtl/xdynamic_bitset.hpp]: Thus we cannot have `xdynamic_bitset_view` on const buffers. This is not blocking for now, but I think we should address this issue sooner or later.

**Comment by JohanMabille** [comment]
Really nice PR, almost ready to be merged. I've added some nitpicking comments, could you also squash a bit, at least to "remove" commits without meaningful comment? No need to squash all, I think it's good to keep the refactoring of the `xdynamic_bitset` and the addition of the `span` class in dedicated commits for instance.

**Comment by wolfv** [comment]
Done!

**Pull request merged**